query/street.js

```diff
@@ -1,13 +1,43 @@
+const SQL = `
+  SELECT * FROM polyline
+  JOIN names ON polyline.id = names.id
+  WHERE polyline.id IN ( %%IDS%% )
+  LIMIT 10
+`;
 
-var SQL = [
-  'SELECT * FROM polyline',
-  'JOIN names ON polyline.id = names.id',
-  'WHERE polyline.id IN ( %%IDS%% )',
-  'LIMIT 10;'
-].join(' ');
+// SQL prepared statements dont easily support variable length inputs.
+// This function dynamically generates a SQL query based on the number
+// of 'id' conditions required.
+function generateDynamicSQL(max) {
+  const conditions = new Array(max.ids)
+    .fill('$id')
+    .map((sql, pos) => sql.replace('$id', `$id${pos}`));
 
-module.exports = function( db, ids, cb ){
+  return SQL.replace('%%IDS%%', conditions.join(','));
+}
+
+// Reusing prepared statements can have a ~10% perf benefit
+// Note: the cache is global and so must be unique per database.
+const cache = [];
+function statementCache(db, max) {
+  const key = `${max.ids}:${db.name}`;
+  if (!cache[key]) {
+    cache[key] = db.prepare(generateDynamicSQL(max));
+  }
+  return cache[key];
+}
+
+module.exports = function( db, ids ){
+  const stmt = statementCache(db, { ids: ids.length });
+
+  // query params
+  const params = {};
+
+  // each name is added in the format: $id0=x, $id1=y
```

Comment:
everything about this comment is confusing to me
- what is x and y
- it says "name" but talks about "ids"
- "added" .. to what?

I think the comment could be

// We generate a prepared statement which can insert multiple polyline ids at once
// the prepared statement has unbound variables id0, id1, id2 and so on
// So generate the binding dictionary with id0 = ids[0], id1 = ids[1] and so on